manticore/core/manticore.py

```diff
@@ -1065,18 +1082,33 @@ def run(self):
         # different process modifies the state
         self.stcache = weakref.WeakValueDictionary()
 
+        if self._introspector is None:
+            self._introspector = self._introspection_plugin()
+            self.register_plugin(self._introspector)
+
         # Lazy process start. At the first run() the workers are not forked.
         # This actually starts the worker procs/threads
         if self.subscribe:
             # User subscription to events is disabled from now on
             self.subscribe = None
 
+        # Passing generators to callbacks is a bit hairy because the first callback would drain it if we didn't
+        # clone the iterator in event.py. We're preserving the old API here, but it's something to avoid in the future.
```

Comment:
That sounds like it would work, although I worry a bit about user callbacks not using the locked APIs to access items on the Manticore object and causing concurrency problems.

Reply:
"I think everything is properly guarded by a lock"
There. I said it.